src/store/TodoListProvider.tsx

```diff
@@ -0,0 +1,42 @@
+import { useState, createContext, ReactNode } from 'react';
+import { getTodoList } from '@todo/api';
+import { TodoItem } from './TodoItem';
+
+type TodoListQuery = {
+  isLoading: boolean;
+  error?: unknown | undefined;
+  todoList: ReadonlyArray<TodoItem>;
+};
+
+type TodoListQueryProps = {
+  queryProps: TodoListQuery;
+  queryTodoList: () => Promise<void>;
+};
+
+const TodoListContext = createContext<TodoListQueryProps>({
+  queryProps: { isLoading: false, error: '', todoList: [] },
+  queryTodoList: async () => {},
+});
+
+const TodoListProvider = ({ children }: { children: ReactNode }) => {
+  const [queryProps, setQueryProps] = useState<TodoListQuery>({
+    isLoading: false,
+    error: '',
```

Comment:
`error: unknown` or you even can remove this line (cause `error` is `nullable` field).

Reply:
Agree, extra variable here.